gpkit/constraints/gp.py

```diff
@@ -305,33 +305,33 @@ def _compile_result(self, solver_out):
         la, self.nu_by_posy = self._generate_nula(solver_out)
         cost_senss = sum(nu_i*exp for (nu_i, exp) in zip(self.nu_by_posy[0],
                                                          self.cost.hmap))
-        self.v_ss = cost_senss.copy()
+        gpv_ss = cost_senss.copy()
         for las, nus, c in zip(la[1:], self.nu_by_posy[1:], self.hmaps[1:]):
             while getattr(c, "parent", None) is not None:
                 c = c.parent
             v_ss, c_senss = c.sens_from_dual(las, nus, result)
             for vk, x in v_ss.items():
-                self.v_ss[vk] = x + self.v_ss.get(vk, 0)
-            while getattr(c, "generated_by", None) is not None:
+                gpv_ss[vk] = x + gpv_ss.get(vk, 0)
```

Comment:
haven't thought deeply about this, but seems like this is acting a lot like a `defaultdict`? Could `gpv_ss` be a default dict, and then this simply becomes `gpv_ss[vk] += x`? Is `defaultdict` noticeably slower?